File: data-tool/src/main/java/com/nilecochen/fitnessapp/datatool/entities/EntryMetric.java
package com.nilecochen.fitnessapp.datatool.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "entry_metrics")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class EntryMetric {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "entry_id", nullable = false)
    private Entry entry;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "metric_id", nullable = false)
    private MetricDefinition metricDefinition;

    @Column(name = "value_number")
    private Double valueNumber;

    @Column(name = "value_text", columnDefinition = "TEXT")
    private String valueText;

    private String unit;
}

File: admin-dash/Admin-Dash/Models/EntryMetric.cs
using System;
using System.Collections.Generic;

namespace Admin_Dash.Models;

public partial class EntryMetric
{
    public long Id { get; set; }

    public long EntryId { get; set; }

    public long MetricId { get; set; }

    public double? ValueNumber { get; set; }

    public string? ValueText { get; set; }

    public string? Unit { get; set; }

    public virtual Entry Entry { get; set; } = null!;

    public virtual MetricDefinition Metric { get; set; } = null!;
}
